pkg/helper/client.go

```diff
@@ -148,8 +147,18 @@ func ApplyDesiredState(nodeNetworkState *nmstatev1alpha1.NodeNetworkState) (stri
 	return setOutput + outputVlanFiltering, nil
 }
 
-func filterOut(currentState nmstatev1alpha1.State, interfacesFilter string) (nmstatev1alpha1.State, error) {
-	if interfacesFilter == "" {
+func getFilter() {
```

Comment:
This should return pointer to the compiled filter. User must not know that a singleton is being used.